acl/src/main/java/org/apache/rocketmq/acl/plain/PlainAccessResource.java

```diff
@@ -281,6 +380,43 @@ public static PlainAccessResource build(PlainAccessConfig plainAccessConfig, Rem
         Permission.parseResourcePerms(plainAccessResource, false, plainAccessConfig.getGroupPerms());
         Permission.parseResourcePerms(plainAccessResource, true, plainAccessConfig.getTopicPerms());
 
+        List<ResourceAndPerm> resourcePerms = plainAccessConfig.getResourcePerms();
+        if (resourcePerms != null && !resourcePerms.isEmpty()) {
+            for (ResourceAndPerm resource : resourcePerms) {
+                String resourceName = resource.getResource();
+                ResourceType type = resource.getType();
+                String namespace = resource.getNamespace();
+                String perm = resource.getPerm();
+                if (type == ResourceType.GROUP) {
+                    String key = namespace == null ? PlainAccessResource.getRetryTopic(resourceName) :
+                        MixAll.RETRY_GROUP_TOPIC_PREFIX + namespace + NAMESPACE_SEPARATOR + resourceName;
+                    plainAccessResource.addResourceAndPerm(key, Permission.parsePermFromString(perm));
+                } else if (type == ResourceType.TOPIC) {
+                    String key = namespace == null ? resourceName : namespace + NAMESPACE_SEPARATOR + resourceName;
+                    plainAccessResource.addResourceAndPerm(key, Permission.parsePermFromString(perm));
+                }
+            }
+        }
+
+        List<NamespaceAndPerm> namespacePerms = plainAccessConfig.getNamespacePerms();
+        if (namespacePerms != null && !namespacePerms.isEmpty()) {
+            Map<String, Map<String, Byte>> namespacePermMap = new HashMap<>();
+            for (NamespaceAndPerm namespace : namespacePerms) {
+                String namespaceName = namespace.getNamespace();
+                String topicPerm = namespace.getTopicPerm();
+                String groupPerm = namespace.getGroupPerm();
+                Map<String, Byte> permMap = new HashMap<>(2);
```

Comment:
Use `com.google.common.collect.Maps#newHashMapWithExpectedSize` to prevent resizing, or `java.util.Collections#singletonMap` if only one entry in this map.